### ISSUE TITLE
Psych-DS and JSON-LD/Schema.org validation should be separate

### ISSUE THREAD

**Issue by bleonar5** [psych-ds/psych-DS#27]
# TL;DR
We propose to create separate, self-contained functions for validation psych-DS vs JSON-LD/Schema.org criteria.

# Detailed View
In order to ensure that our system is sufficiently modular and extensible, we propose that psych-DS and JSON-LD/Schema.org criteria should be validated by separate, self contained components. Part of the reasoning for this is that we predict that changes could arise in either set of criteria at any point, and that these changes will arise independently, since the two standards are managed by different organizations, and so it would make sense to separate the structures in which those criteria are stored/referenced by the app. Naturally, our goal will be to integrate the modules to the maximum extent possible, sharing as much schema-agnostic functionality and architecture as possible.

**Comment by mekline** [comment]
Another point: the rules that Psych-DS metadata enforces over & above schema.org metadata can be thought of as the "base extension" for Psych-DS - if communities build on top of Psych-DS in the future, they would be adding additional extensions in a similar way - e.g. by adding rules such as "variableMeasured must include `age`" or "countryOfOrigin must be present and must be a 3-letter ISO country code"
